### PR TITLE
written assignment file directory naming

### PR DESCRIPTION
**Pull request by TheMaja**
- introduction to written assignements
- project added

Related issues: [please specify]

## Description:

What are you up to? Fill us in :)

## I solemnly swear that:

- [x] I ran the hugo server and looked at my changed in the browser with my own eyes
- [x] I ran the linter and there were no errors
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]
Mostly looks good to me, just a few questions about loaded questions below.

File: content/syllabus-written-assignments/file-directory-naming/_index.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+This assessment is designed to not only assess your current knowledge but also provide insights into areas where you can further improve. It is an integral part of your learning journey, contributing to your growth as a capable and confident programmer.
+
+## Questions
+
+1. What is the recommended naming convention for files and directories in the programming language you are learning? Why is it important to follow this convention?
+
+2. What are some of the common mistakes that beginners make when naming files and directories? How can you avoid these mistakes?
+
+3. Explain the importance of using naming conventions when organizing files and directories. How do naming conventions help with locating, identifying, and retrieving information from a file system? Provide an example of a specific naming convention that you might use for categorizing different types of files.
```
Comment (Ngoakor12) [inline content/syllabus-written-assignments/file-directory-naming/_index.md]: The questions are loaded. I think it would be good if each question had a single and not a multipart question because learners are likely to answer some parts and leave other parts for a question.

1. the question can be broken down into 2. one about naming conventions and the other about the important of naming conventions
2. I'm not sure if the second question is necessary if the answer is like just going to be "follow naming conventions"
3. This one can also be broken down further

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]

File: content/syllabus-written-assignments/file-directory-naming/_index.md [hunk 2]
```diff
@@ -15,11 +15,13 @@ This assessment is designed to not only assess your current knowledge but also p
...
-1. What is the recommended naming convention for files and directories in the programming language you are learning? Why is it important to follow this convention?
+1. What naming convention is recommended for files and directories in the programming language you're learning?
 
-2. What are some of the common mistakes that beginners make when naming files and directories? How can you avoid these mistakes?
+2. Why is it of importance to apply naming conventions when arranging files and directories?
 
-3. Explain the importance of using naming conventions when organizing files and directories. How do naming conventions help with locating, identifying, and retrieving information from a file system? Provide an example of a specific naming convention that you might use for categorizing different types of files.
+3. How do naming conventions help with locating, identifying, and retrieving information from a file system?
+
+4. Provide an example of a specific naming convention that you might use for categorizing different types of files.
```
Comment (Ngoakor12) [inline content/syllabus-written-assignments/file-directory-naming/_index.md]: These are much lighter for sure, but I wouldn't know how to answers them. Can you rephrase them a bit if possible.

File: content/syllabus-written-assignments/file-directory-naming/_index.md [hunk 3]
```diff
@@ -15,11 +15,13 @@ This assessment is designed to not only assess your current knowledge but also p
 
 ## Questions
 
-1. What is the recommended naming convention for files and directories in the programming language you are learning? Why is it important to follow this convention?
+1. What naming convention is recommended for files and directories in the programming language you're learning?
 
-2. What are some of the common mistakes that beginners make when naming files and directories? How can you avoid these mistakes?
+2. Why is it of importance to apply naming conventions when arranging files and directories?
```
Comment (Ngoakor12) [inline content/syllabus-written-assignments/file-directory-naming/_index.md]: How about rephrasing this to: `Why is it important to use recommended naming conventions?`

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]

**Review by Ngoakor12 — APPROVED** [review]
Looks good to me

File: content/syllabus-written-assignments/file-directory-naming/_index.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+---
+content_type: project
+flavours:
+  - markdown
```
Comment (sheenarbw) [inline content/syllabus-written-assignments/file-directory-naming/_index.md]: it would be useful to include the language flavors as well. Otherwise this wont be abler to be automarked.

File: content/syllabus-written-assignments/file-directory-naming/_index.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ready: true
+submission_type: repo
+title: File and Directory Naming
+---
+
+This assessment is designed to not only assess your current knowledge but also provide insights into areas where you can further improve. It is an integral part of your learning journey, contributing to your growth as a capable and confident programmer.
+
+## Questions
+
+1. What naming convention is recommended for files and directories in the programming language you're learning?
```
Comment (sheenarbw) [inline content/syllabus-written-assignments/file-directory-naming/_index.md]: I would separate this out into multiple questions so that they can be automarked. 

Eg: 

1. Here are some variable names that follow different naming conventions: Foo_Bar, foo_bar, FooBar,  fooBar. Which of the variables follows the convention used in your language? 
2. Here are some directory names...
3. ...function names...

File: content/syllabus-written-assignments/file-directory-naming/_index.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+2. Why is it important to use the recommended naming convention?
+
+3. How does using clear names make it easier to locate and recognize files and folders in a directory?
```
Comment (sheenarbw) [inline content/syllabus-written-assignments/file-directory-naming/_index.md]: Question 3 seems to be a partial answer for question 2.  It might be good to remove it or to remove redundancy in another way.

Question 4 also doesn't seem useful given question 1.

File: content/syllabus-written-assignments/file-directory-naming/_index.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+prerequisites:
+  hard:
+    - topics/file-and-directory-naming/general
+protect_main_branch: false
+ready: true
+submission_type: repo
+title: File and Directory Naming
+---
+
+This assessment is designed to not only assess your current knowledge but also provide insights into areas where you can further improve. It is an integral part of your learning journey, contributing to your growth as a capable and confident programmer.
```
Comment (sheenarbw) [inline content/syllabus-written-assignments/file-directory-naming/_index.md]: This paragraph seems a bit fluffy. It has a lot of words but it doesn't say much.

Surely all required topics and projects are an integral part of the learning journey? Surely all projects are designed to assess knowledge and provide insights about where the learner can improve?  

I'm sure this whole paragraph can be replaced with: "Please answer the following questions" or something like that.

**Review by sheenarbw — CHANGES_REQUESTED** [review]
The questions can be better.

File: content/syllabus-written-assignments/file-directory-naming/_index.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+title: File and Directory Naming
+---
+
+Please answer the following questions in markdown files and submit following the instructions below.
+
+## Questions
+
+1. Here are some directory names that follow different naming conventions: Foo_Bar, foo_bar, FooBar, main, fooBar. Which of the directory names follows the convention used in your language?
+
+2. Here are some file names that follow different naming conventions: MyFile my_file My_File. Which of the file names follows the convention used in your language?
```
Comment (Ngoakor12) [inline content/syllabus-written-assignments/file-directory-naming/_index.md]: - separate the examples with commas.

Comment (Ngoakor12) [reply content/syllabus-written-assignments/file-directory-naming/_index.md]: - You are also missing examples for function names?

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]

**Review by Ngoakor12 — APPROVED** [review]

File: content/syllabus-written-assignments/file-directory-naming/_index.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+ready: true
+submission_type: repo
+title: File and Directory Naming
+---
+
+Please answer the following questions in markdown files and submit following the instructions below.
+
+## Questions
+
+1. Here are some directory names that follow different naming conventions: Foo_Bar, foo_bar, FooBar, main, fooBar. Which of the directory names follows the convention used in your language?
```
Comment (sheenarbw) [inline content/syllabus-written-assignments/file-directory-naming/_index.md]: why is main in there? It'll be much easier to auto mark if there is only one correct answer.

File: content/syllabus-written-assignments/file-directory-naming/_index.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+2. Here are some file names that follow different naming conventions: MyFile, my_file, My_File. Which of the file names follows the convention used in your language?
+
+3. Here are function names that follow different naming conventions: myFunction, my_function, MyFunction. Which of the function names follows the convention used in your language?
```
Comment (sheenarbw) [inline content/syllabus-written-assignments/file-directory-naming/_index.md]: Add some crazy ones like My_function My_Function. People do strange mix-and-match like that often enough that it's worth discouraging

File: content/syllabus-written-assignments/introduction/_index.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
+---
+content_type: topic
+ready: true
+title: Introduction to written assignments
+---
+
+## Intro
+
+We are adding written assignments as an important part of your learning experience. These assignments are designed to help you learn better and overcome common challenges.
```
Comment (sheenarbw) [inline content/syllabus-written-assignments/introduction/_index.md]: They don't need to know that we are still busy adding writing assignments.

File: content/syllabus-written-assignments/introduction/_index.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+
+## Why
+
+Sometimes, just reading content might not be enough to fully understand certain topics. So, we want to make sure you can apply what you've learned and express your understanding clearly.
+
+## Outcomes
+
+By completing these written assignments, you'll be more engaged with the course material, and it will be easier for you to remember what you've learned.
```
Comment (sheenarbw) [inline content/syllabus-written-assignments/introduction/_index.md]: The learners will probably ask: "What writing assignments?" .

And aren't all of these outcomes really just reasons? Why are some reasons under "why" and others are under "outcomes"?

File: content/syllabus-written-assignments/introduction/_index.md [hunk 13]
```diff
@@ -0,0 +1,19 @@
+---
+content_type: topic
```
Comment (sheenarbw) [inline content/syllabus-written-assignments/introduction/_index.md]: This file doesn't seem to be referenced anywhere. That means nobody will ready it.

I'm not sure that it adds value anyway. It would need a bit of work before we can include it. To speed things up I'd recommend removing it from the  PR. Maybe make a separate issue about introducing learners to writing assignments. But I'm not sure if they need an introduction or not.

File: content/syllabuses/data-eng-part-1.md [hunk 14]
```diff
@@ -7,6 +7,7 @@ _db_id: 4
 - {{< contentlink path="agile/agile-scrum" >}}
 - {{< contentlink path="topics/clean-code/python" >}}
 - {{< contentlink path="topics/file-and-directory-naming/python" >}}
+- {{< contentlink path="syllabus-written-assignments/file-directory-naming" >}}
```
Comment (sheenarbw) [inline content/syllabuses/data-eng-part-1.md]: Once upon a time we separated everything out into topic and project directories because we didn't have good frontmatter yet. That is technical debt that needs to be overcome. We shouldn't imitate technical debt.

Syllabus content should be thought of as code. Good code should be cohesive. 

The file and directory naming assignment should be with the file and directory naming content. It shouldn't be super far away. The things that are used together should be together.

**Review by sheenarbw — CHANGES_REQUESTED** [review]
There are still a few things to fix. I also noticed that you're imitating technical debt (therefore creating more technical debt)

File: content/file-and-directory-naming/file-directory-naming-assignment/_index.md [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+
+3. Here are function names that follow different naming conventions: myFunction, my_function, MyFunction, My_function, My_Function. Which of the function names follows the convention used in your language?
+
+4. Why is it important to use the recommended naming convention?
+
+## How to submit your work
+
+Please follow the following instructions to submit your work:
+
+{{< contentlink path="project-submission-instructions/markdown-questions" >}}
```
Comment (Ngoakor12) [inline content/file-and-directory-naming/file-directory-naming-assignment/_index.md:33]: This link gets rendered like this:
image

I assume it's a problem. See if you can't find a way to remove that, otherwise creating a separate ticket for this sort of issue might be the way to go.

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]
previous changes addressed

but one thing...

File: content/file-and-directory-naming/file-directory-naming-assignment/_index.md (hunk 15 above)
Comment (TheMaja) [reply content/file-and-directory-naming/file-directory-naming-assignment/_index.md:33]: @Ngoakor12 it will need a separate ticket, I have seen that [TODO] it many other places.

**Review by Ngoakor12 — APPROVED** [review]

File: content/file-and-directory-naming/java/_index.md [hunk 16]
```diff
@@ -12,10 +12,11 @@ title: File and directory naming in Java
 ---
 
 ## How should I name my Java files and directories?
-When you think about how to name files and directories in Java, you should know that different Java programmers can have different styles and approaches to the way they program. When naming stuff you must use Java standard conventions that are readable and meaningful to yourself and other programmers.
 
+When you think about how to name files and directories in Java, you should know that different Java programmers can have different styles and approaches to the way they program. When naming stuff you must use Java standard conventions that are readable and meaningful to yourself and other programmers.
```
Comment (sheenarbw) [inline content/file-and-directory-naming/java/_index.md]: Is this not true for all languages? Do we already mention this in the general section?

All languages allow coders to be chaotic in how they name things
All languages have certain best practices and conventions around naming
All serious projects adopt a naming convention
All good programmers aim to be clear and consistent by making use of clear conventions instead of being chaotic

Another thing worth mentioning is that people have a bad habit of writing crazy horrible code, then trying to clean up the names after they get the code to work. This is seriously bad because it can introduce bugs, and the coder will often miss something. It's much better just to develop good habits from the start. It'll slow you down at first, but then you will move much faster when you get used to it.

File: content/file-and-directory-naming/java/_index.md [hunk 17]
```diff
@@ -12,10 +12,11 @@ title: File and directory naming in Java
...
 
 ### Directories
+
 Directory names in Java should be in lower case which means that you only use small letters. Eg `src` or `main`
```
Comment (sheenarbw) [inline content/file-and-directory-naming/java/_index.md]: It would be worth saying that if you are using a tool like Gradle to set up your project structure, then just go with the conventions it sets up.

File: content/file-and-directory-naming/javascript/_index.md [hunk 18]
```diff
@@ -29,4 +29,4 @@ Students are expected to be on the same page when it comes to naming their files
     >src
```
Comment (sheenarbw) [inline content/file-and-directory-naming/javascript/_index.md]: This looks a bit inconsistent. How do we usually describe directory hierarchies?

**Review by sheenarbw — CHANGES_REQUESTED** [review]
Some small things

File: content/file-and-directory-naming/general/_index.md [hunk 19]
```diff
@@ -0,0 +1,23 @@
+---
+_db_id: 768
+content_type: topic
+ready: true
+title: File and directory naming
+---
+
+## Naming files and directories
+
+People often develop a habit of writing messy and unorganized code initially, only to attempt to improve the naming conventions after the code functions. This practice is problematic as it can introduce errors, and developers may overlook important details. It is far more advantageous to cultivate good coding habits from the outset. While it may seem slower initially, you will ultimately achieve greater efficiency once you become accustomed to it.
```
Comment (sheenarbw) [inline content/file-and-directory-naming/general/_index.md]: Can you rephrase this? Try to keep things conversational instead of overly stuffy and formal. Think about the audience. 

Also, read this: https://socialtriggers.com/the-big-problem-with-big-words-hint-they-make-you-look-stupid/ 

A lot of people think that if they use fancy words then they are fancy people, it's pretty counterproductive most of the time.

File: content/file-and-directory-naming/general/_index.md [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+
+You can organize your files and directories in a logical and consistent manner by using naming conventions. The use of naming conventions helps to ensure the correct information can be located, identified, and retrieved from a file system quickly, as well as that it is stored in a secure location.
```
Comment (sheenarbw) [inline content/file-and-directory-naming/general/_index.md:10]: This paragraph is not related to the one above. Should it be in the same section? 

Maybe there should be a heading that says "## A common pitfall" or something like that at the bottom that tells people about the common problem.

And then it would be good to say something about why naming matters near the top somewhere.

File: content/file-and-directory-naming/java/_index.md [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+ 
+ > MustBeMeangingful.java
+
+```
+
+### Example of what's expected
+
+```
+
+├── src
```
Comment (sheenarbw) [inline content/file-and-directory-naming/java/_index.md:47]: Sometimes you use this format, and sometimes you use these `>`

Can you please aim to be consistent?

Please also remove all the extra whitespace. I'm not sure why there are so many blank lines in this file.

Hugo will tidy things up when it converts things into markdown. But it would be good for you to develop the habit of being tidy.  It would be nice if the markdown was formatted in a consistent way. You do know a little bit about the benefits of consistency?

**Review by sheenarbw — CHANGES_REQUESTED** [review]
Still a few issues.

File: content/file-and-directory-naming/general/_index.md [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+
+## What should I consider when naming files and directories
+
+New files and directories (including third-party frameworks) should adhere to the recommended naming conventions, but if a library already exists that
+has a different style, internal consistency should take precedence.
+
+## A common pitfall to avoid
+
+Many times, people start by writing code that's messy and not very organized. They only try to make it look neater after it works. But this can cause problems because it might make mistakes, and they might miss important things. It's better to start with good coding habits from the beginning, even if it seems a bit slower at first. Once you get used to it, you'll end up being better and faster at it.
```
Comment (Ngoakor12) [inline content/file-and-directory-naming/general/_index.md]: the 3rd sentence is a bit confusing, I think it's a mistake? Should it not be: `But this can cause problems because they might make mistakes, and they might miss important things`

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]

**Review by Ngoakor12 — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]
